### ISSUE TITLE
/edit Fails with "[Errno 32] Broken pipe"

### ISSUE THREAD

**Issue by d-imal** [continuedev/continue#301]
`/help` commands get a response, so I assume I'm hitting the server. But even the simplest `/edit` commands fail. I'm editing a TypeScript file, if that matters. 

Here's the traceback:

```
Traceback (most recent call last):

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 490, in wait_for
    return fut.result()
           ^^^^^^^^^^^^

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/futures.py", line 198, in result
    raise exc

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 269, in __step
    result = coro.throw(exc)
             ^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/libs/util/queue.py", line 17, in get
    return await self.queues[message_type].get()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/queues.py", line 158, in get
    await getter

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/futures.py", line 287, in __await__
    yield self  # This tells Task to wait for completion.
    ^^^^^^^^^^

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 339, in __wakeup
    future.result()

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/futures.py", line 198, in result
    raise exc

asyncio.exceptions.CancelledError


The above exception was the direct cause of the following exception:


Traceback (most recent call last):

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/server/ide.py", line 166, in _receive_json
    return await asyncio.wait_for(self.sub_queue.get(message_type), timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/Cellar/python@3.11/3.11.2_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 492, in wait_for
    raise exceptions.TimeoutError() from exc

TimeoutError


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/autopilot.py", line 244, in _run_singular_step
    observation = await step(self.continue_sdk)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/main.py", line 320, in __call__
    return await self.run(sdk)
           ^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/plugins/steps/core/core.py", line 606, in run
    file_contents = await sdk.ide.readRangeInFile(range_in_file)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/server/ide.py", line 367, in readRangeInFile
    full_contents = await self.readFile(range_in_file.filepath)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/server/ide.py", line 343, in readFile
    resp = await self._send_and_receive_json({
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/server/ide.py", line 173, in _send_and_receive_json
    resp = await self._receive_json(message_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/server/ide.py", line 168, in _receive_json
    raise Exception(

Exception: ('IDE Protocol _receive_json timed out after 20 seconds', 'readFile')


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/autopilot.py", line 244, in _run_singular_step
    observation = await step(self.continue_sdk)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/main.py", line 320, in __call__
    return await self.run(sdk)
           ^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/plugins/steps/main.py", line 258, in run
    await sdk.run_step(DefaultModelEditCodeStep(user_input=self.user_input, range_in_files=range_in_files))

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/sdk.py", line 203, in run_step
    return await self.__autopilot._run_singular_step(step)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/Users/danimal/.continue/server/env/lib/python3.11/site-packages/continuedev/core/autopilot.py", line 261, in _run_singular_step
    print(

BrokenPipeError: [Errno 32] Broken pipe
```

**Comment by sestinj** [comment]
Thanks for the traceback! If you're intent on using Continue, a band-aid solution _might_ be to `lsof -i :65432`, kill the "(LISTENER)" PID with `kill <PID>`, and then restart VS Code. But I'll get back soon with a real fix

**Comment by d-imal** [comment]
Thanks. I'm dealing with #305 now, though. I left it running and walked away yesterday. Came back this morning and it's still going...

**Comment by d-imal** [comment]
Trying to view logs in VS Code gave me this error: "Command 'Continue: View Logs' resulted in an error. command 'continue.viewLogs' not found"

**Comment by sestinj** [comment]
@d-imal thanks for the update. I'm going to move this over to #305 and close this issue. As for the viewing logs error, I had been registering commands only after the server loaded. I just fixed this in the newest version (v0.0.222), so you will be able to see the logs.
